android/src/main/java/com/reactnativestripesdk/pushprovisioning/TapAndPayProxy.kt

```diff
@@ -2,54 +2,68 @@ package com.reactnativestripesdk.pushprovisioning
 
 import android.app.Activity
 import android.util.Log
-import com.facebook.react.bridge.Promise
 import com.facebook.react.bridge.ReadableMap
 import com.facebook.react.bridge.WritableMap
 import com.facebook.react.bridge.WritableNativeMap
 import com.reactnativestripesdk.createError
 import com.google.android.gms.tasks.Task
 
+typealias TokenCheckHandler = (isCardInWallet: Boolean, token: WritableMap?, error: WritableMap?) -> Unit
+
 object TapAndPayProxy {
   private const val TAG = "StripeTapAndPay"
   private var tapAndPayClient: Any? = null
   const val REQUEST_CODE_TOKENIZE = 90909
 
-  fun invoke(activity: Activity, newCardLastFour: String, promise: Promise) {
-    try {
+  private fun getTapandPayTokens(activity: Activity): Task<List<Any>>? {
+    return try {
       val tapAndPayClass = Class.forName("com.google.android.gms.tapandpay.TapAndPay")
-      val getClientMethod = tapAndPayClass.getMethod("getClient", Activity::class.java)
+      val getClientMethod = tapAndPayClass.getMethod(
+        "getClient",
+        Activity::class.java)
+      val client = getClientMethod.invoke(null, activity)
 
-      tapAndPayClient = getClientMethod.invoke(null, activity).also {
-        val tapAndPayClientClass = Class.forName("com.google.android.gms.tapandpay.TapAndPayClient")
-        val listTokensMethod = tapAndPayClientClass.getMethod("listTokens")
+      val tapAndPayClientClass = Class.forName("com.google.android.gms.tapandpay.TapAndPayClient")
```

Comment:
Can you depend on the library as `compileOnly` so that you don't need to use reflection?

Reply:
No, unfortunately since google's library is distributed as an .aar, we cannot use compileOnly 😞